JSONLibrary/jsonlibrary.py

```diff
@@ -0,0 +1,318 @@
+import os.path
+import jsonschema
+from copy import deepcopy
+from robot.api import logger
+from robot.utils.asserts import fail
+from jsonpath_ng import Index, Fields
+from jsonpath_ng.ext import parse as parse_ng
+from jsonpath_ng.exceptions import JsonPathParserError
+
+__author__ = "Traitanit Huangsri"
+__email__ = "traitanit.hua@gmail.com"
+
+
+class JSONLibrary:
+    """JSONLibrary is a robotframework testlibrary for manipulating JSON object (dictionary)
+
+    You can get, add, update and delete your json object using JSONPath.
+
+    == JSONPath Syntax ==
+    | JSONPath | Description |
+    | $        | the root object/element |
+    | @        | the current object/element |
+    | . or []  | child operator |
+    | ..       | recursive descent. JSONPath borrows this syntax from E4X |
+    | *        | wildcard. All objects/element regardless their names. |
+    | []       | subscript operator. XPath uses it to iterate over element collections and for predicates.
+                 In Javascript and JSON it is the native array operator. |
+    | [,]      | Union operator in XPath results in a combination of node sets. JSONPath allows alternate
+                 names or array indices as a set. |
+    | [start:end:step] | array slice operator borrowed from ES4 |
+    | ?()      | applies a filter (script) expression. |
+    | ()       | script expression, using the underlying script engine. |
+
+    == *** Known issue *** ==
+    If there is a space in JSONPath expression, the module used by this library will throw an exception.
+    Therefore, please avoid the space in JSONPath expression if possible.
+
+    *Example:*
+    | JSONPath | Exception? |
+    | $.[?(@.id == 1)] | Y |
+    | $.[?(@.id==1)] | N |
+    | $.[?(@.name=='test 123')] | N |
+
+    == Example Test Cases ==
+    | *** Settings ***     |
+    | Library              | JSONLibrary |
+    |                      |
+    | *** Test Cases ***   |
+    | TestManipulatingJSON |
+    | ${json_object}=      |  Load JSON From File  |  example.json  |
+    | ${object_to_add}=    |  Create Dictionary    |  country=Thailand |
+    | ${json_object}=      |  Add Object To Json   |  ${json_object}  |  $..address  | ${object_to_add} |
+    | ${value}=            |  Get Value From Json  |  ${json_object}  |  $..country  |
+    | Should Be Equal As Strings  |  ${value[0]}   | Thailand  |
+
+
+    """
+
+    ROBOT_LIBRARY_SCOPE = "GLOBAL"
+    ROBOT_LIBRARY_DOC_FORMAT = "ROBOT"
+    ROBOT_EXIT_ON_FAILURE = True
+
+    @staticmethod
+    def _parse(json_path):
+        try:
+            return parse_ng(json_path)
+        except JsonPathParserError as e:
+            fail(
+                "Parser failed to understand syntax '{}'. error message: "
+                "\n{}\n\nYou may raise an issue on https://github.com/h2non/jsonpath-ng".format(
+                    json_path, e
+                )
+            )
+
+    @staticmethod
+    def load_json_from_file(file_name, encoding=None):
```

Comment:
I know this IDE bug, this bug only occurs when using `@keyword` with `'`

The decorator `@keyword` in the previous implementation is not necessary, because the the keyword name is the same with the method, IDE will understand the method name safely.